### PR TITLE
[LEC] Create ExecutionCorrectness service

### PR DESCRIPTION
**Pull request by lightmark**
## Motivation
2 commits:

1 . We need to move the traits to executor_types to tidy up the dependencies.
Also, a customized `Error` type is necessary for sending results over wire for the future use of this trait in execution correctness service.

2. Create the ExecutionCorrectness Service (LEC service) similar to safety-rules. No asymmetric cryptography involved yet.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

CI

### CODE REVIEW

File: execution/execution-correctness/src/spawned_process.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                        "Spawned ExecutionCorrectness process could not be killed: '{}'",
+                        e
+                    );
+                }
+            }
+        }
+    }
+}
+
+#[cfg(test)]
```
Comment (davidiw) [inline execution/execution-correctness/src/spawned_process.rs:61]: why not use workspace-builder?

File: execution/executor-types/src/lib.rs [hunk 2]
```diff
@@ -27,7 +82,7 @@ use storage_interface::TreeState;
 /// of success / failure of the transactions.
 /// Note that the specific details of compute_status are opaque to StateMachineReplication,
 /// which is going to simply pass the results between StateComputer and TxnManager.
-#[derive(Debug, Default, PartialEq, Eq, Clone)]
+#[derive(Debug, Default, PartialEq, Eq, Clone, Serialize, Deserialize)]
```
Comment (davidiw) [inline execution/executor-types/src/lib.rs:85]: let's order this!

File: execution/executor-utils/src/lib.rs [hunk 3]
```diff
@@ -18,7 +18,7 @@ pub fn create_storage_service_and_executor(
     let (db, db_rw) = init_libra_db(config);
     bootstrap_db_if_empty::<LibraVM>(&db_rw, get_genesis_txn(config).unwrap()).unwrap();
 
-    let _handble = start_simple_storage_service_with_db(config, db.clone());
+    let _handle = start_simple_storage_service_with_db(config, db.clone());
```
Comment (davidiw) [inline execution/executor-utils/src/lib.rs:21]: haha

File: types/src/transaction/mod.rs [hunk 4]
```diff
@@ -582,7 +582,7 @@ impl From<TransactionWithProof> for crate::proto::types::TransactionWithProof {
 /// The status of executing a transaction. The VM decides whether or not we should `Keep` the
 /// transaction output or `Discard` it based upon the execution of the transaction. We wrap these
 /// decisions around a `VMStatus` that provides more detail on the final execution state of the VM.
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
```
Comment (davidiw) [inline types/src/transaction/mod.rs:585]: order plz

File: execution/execution-correctness/src/execution_correctness_manager.rs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+use executor_types::BlockExecutor;
+use libra_config::config::{ExecutionCorrectnessService, NodeConfig};
+use libra_vm::LibraVM;
+use simple_storage_client::SimpleStorageClient;
+use std::{
+    net::SocketAddr,
+    sync::{Arc, Mutex},
+};
+
+enum ExecutionCorrectnessWrapper {
```
[the remark this answers is not in the document]
Comment (lightmark) [reply execution/execution-correctness/src/execution_correctness_manager.rs:22]: Zekun knows.

File: execution/execution-correctness/src/spawned_process.rs (hunk 1 above)
Comment (lightmark) [reply execution/execution-correctness/src/spawned_process.rs:61]: What do you mean here?

Comment (davidiw) [reply execution/execution-correctness/src/spawned_process.rs:61]: I misread, please ignore...

**Pull request closed without merging**